xcdat/dataset.py

```diff
@@ -35,13 +37,97 @@
 ]
 
 
-def open_dataset(
+def open(
     path: str,
     data_var: Optional[str] = None,
     add_bounds: bool = True,
     decode_times: bool = True,
     center_times: bool = False,
     lon_orient: Optional[Tuple[float, float]] = None,
+    **kwargs,
+) -> xr.Dataset:
+    """
+    Documentation to be written when this PR is
+    further along...
+    """
+
+    # inspect path attributes and use either open_dataset
+    # or open_mfdataset as appropriate
+    #   * list (open list of files with open_mfdataset)
+    #   * pathlib.Path (open pathlib.Path with open_mfdataset)
+    #   * deprecated CDAT xml file type
```

Comment:
A few questions I had about supporting XML:
- I noticed in the docstring it says "deprecated CDAT xml file type". Is the structure of the XML formatted specifically by CDAT, and are there plans to eventually move away from it?
  - I think if users pass XML files that don't align with a specific format, then `open` will break. We'd have to include documentation that the XML file must follow a specific format (not sure if this is appropriate to support though since it isn't generalized like a list of paths).
- I'm not really familiar with XML usage. How common is the XML format used for reading datasets in the community?

Reply:
* The structure of CDAT xml files (climate data markup language) is a [dialect of xml files](https://cdms.readthedocs.io/en/latest/manual/cdms_6.html) that have a defined set of attributes. 
* We only care that the xml file has a `directory` attribute – we can probably just return an error saying "the xml file provided does not have a root directory attribute" or something like that
* XML probably isn't used that widely, but it is used pretty widely by former CDAT users

Reply:
Gotcha, this makes sense to me. I think being explicit about these details as you describe is a good idea.